dropbox.go

```diff
@@ -0,0 +1,71 @@
+package filecache
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/dropbox/dropbox-sdk-go-unofficial/dropbox"
+	"github.com/dropbox/dropbox-sdk-go-unofficial/dropbox/files"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/oauth2"
+)
+
+// DropboxDownload will fetch a file from the specified Dropbox path into a localPath. It
+// will create sub-directories as needed inside that path in order to store the
+// complete path name of the file.
+func DropboxDownload(file File, localPath string, downloadTimeout time.Duration) error {
+	// Make sure the path to the local file exists
+	err := os.MkdirAll(filepath.Dir(localPath), 0755)
```

Comment:
All of this code that is more or less shared with AWS should be pulled out into a func.